src/lib/evil/evil_libgen.h

```diff
@@ -0,0 +1,13 @@
+#ifndef __EVIL_LIBGEN_H__
+#define __EVIL_LIBGEN_H__
+
+#include <evil_api.h>
```

Comment:
You should not include `evil_api.h` except in `evil_private.h`. You can think that `evil_private.h` is "the lib", the others are just "organized parts of it".

So no one includes `evil_{something that is not 'private'}.h`, but `evil_private.h` includes everyone. Meanwhile no `evil_*` header includes any other `evil_*` (not even `evil_private.h`), every other file should include `evil_private.h` if it wants to use `evil`.

Reply:
If I understood right, you are saying that we should be doing [this](https://github.com/expertisesolutions/efl/commit/c7389b77c9693c6af595ab55fae4e8381b22d675)?

Reply:
yes, and no. In c7389b77c9693c6af595ab55fae4e8381b22d675 there are still some `evil_<thing>.h` that are including other `evil` headers besides `evil_private.h`. Accordingly to @JPTIZ they if they need anything from other `evil`, they should only include `evil_private.h`.

Reply:
Nice, checkout 137a55a0ff4683c6420902c8e8934362d7454a99.